corehq/apps/reports/standard/users/reports.py

```diff
@@ -179,15 +180,39 @@ def _html_list(changes, unstyled=True):
 
     properties = UserHistoryReport.get_primary_properties(domain)
     properties.pop("user_data", None)
-    primary_changes = {
-        properties.get(key, key): value for key, value in changes.items()
-        if key in properties
-    }
-    more_count = len(changes) - len(primary_changes)
+    primary_changes = {}
+
+    for key, value in changes.items():
+        if key in properties:
+            if key == 'commcare_location_id':
+                primary_changes[_("primary location (mobile users only)")] = cached_location_id_to_name(value)
+            if key == 'location_id':
+                primary_changes["location"] = cached_location_id_to_name(value)
+            else:
+                primary_changes[key] = value
```

Comment:
```suggestion
                primary_changes[properties[key]] = value
```
This current behavior drops the mapping from one to the other that was in the old version.

In fact, I think a simpler way of writing this would be to add `commcare_location_id` to the dict at the top, so that stores all the key mappings:
```python
'commcare_location_id': _("primary location (mobile users only)")
```

Then this block could transform only the values:
```python
    for key, value in changes.items():
        if key in properties:
            if key in ('commcare_location_id', 'location_id'):
                value = cached_location_id_to_name(value)
            primary_changes[properties[key]] = value
```

Reply:
Thanks @esoergel, I had not seen that doc on caching. I'll read over it. It does seem like the `@memoized` decorator is the most appropriate way to go.

Reply:
I think you replied to a different comment here.  Did you see this one?  I think this behavior change was made in error.

Reply:
Thanks Ethan, exactly, right. Replied to the wrong comment. I'll look at your suggestion about managing key mapping and value changes first thing on Tuesday.